sklearn/tests/test_common.py

```diff
@@ -332,14 +332,33 @@ def test_check_n_features_in_after_fitting(estimator):
     check_n_features_in_after_fitting(estimator.__class__.__name__, estimator)
 
 
+def _estimators_that_predict_in_fit():
+    for estimator in _tested_estimators():
+        est_params = set(estimator.get_params())
+        if "oob_score" in est_params:
+            yield estimator.set_params(oob_score=True, bootstrap=True)
+        elif "early_stopping" in est_params:
+            est = estimator.set_params(early_stopping=True, n_iter_no_change=1)
+            if est.__class__.__name__ in {"MLPClassifier", "MLPRegressor"}:
+                # TODO: FIX MLP to not check validation set during MLP
+                yield pytest.param(
+                    est, marks=pytest.mark.xfail(msg="MLP still validates in fit")
+                )
```

Comment:
I noticed that the original tests did not cover all the estimators that uses a validation set. I updated it and saw that MLP still validates the validation set during `fit`. The fix to MLP is a little more involved and I do not want to block this PR. I marked the test here to xfail for the MLP case.

I will followup with a PR for MLP.